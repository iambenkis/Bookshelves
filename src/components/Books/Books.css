img {
  width: 100%;
  height: 100%;
  margin-bottom: 0.5rem;
}

.books {
  display: flex;
  flex-wrap: wrap;
  padding: 0.1rem 0.5rem;
}

.book-item {
  border: 1px solid rgb(189, 189, 189);
  width: 49%;
  margin-right: 0.3rem;
  margin-bottom: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.item-details {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.7rem 0.5rem;
  color: #fff;
}

.chevron {
  cursor: pointer;
  color: #fff;
}
