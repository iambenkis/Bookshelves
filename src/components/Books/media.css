@media (min-width : 920px) {
.book-item {
    border: 1px solid rgb(189, 189, 189);
    width: 15%;
    margin-right: 1rem;
    margin-bottom: 1rem;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.books {
    align-items: center;
    justify-content: center;
    width: 100%;
}
}