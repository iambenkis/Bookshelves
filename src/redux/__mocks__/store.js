function store() {
  return [
    {
      id: 0,
      review: 3.6,
      description: 'THE AMAZING NEW THRILLER FROM TERRY HAYES.',
      Source: 'www.amazon.co.uk',
      title: 'The Year of the Locust: Terry Hayes',
    },
  ];
}

export default store;
