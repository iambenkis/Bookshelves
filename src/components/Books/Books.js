import React from 'react';
import { useSelector } from 'react-redux';
import { FaChevronCircleRight } from 'react-icons/fa';
import './Books.css';
import './media.css';
import { Link } from 'react-router-dom';
import StarRating from '../StartRating/StartRating';

const Books = () => {
  const bookStore = useSelector((state) => state.books);
  const { Books } = bookStore;
  const regex = /[+-]?\d+(\.\d+)?/g;
  const getStars = (star) => {
    const nbrStart = star.match(regex)[0];
    return Number(nbrStart);
  };
  return (
    <div className="book-container">
      <div className="books">
        {
                    Books?.filter((b) => b.id < 16).map((book) => (
                      <div className="book-item" key={book.id}>
                        <img src={book.imgUrl} alt="img" />
                        <div className="item-details">
                          <StarRating star={getStars(book.review)} />
                          <p>{book.price}</p>
                          <Link className={({ isActive }) => `link ${(isActive ? 'active' : '')}`} to={`/detail/${book.id}`}>
                            <FaChevronCircleRight
                              size={15}
                              className="chevron"
                            />
                          </Link>
                        </div>
                      </div>
                    ))
                }
      </div>
    </div>
  );
};

export default Books;
