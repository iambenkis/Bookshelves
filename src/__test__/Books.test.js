import React from 'react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
// import store from '../redux/Rockets/configurationStore';
// import Rocket from '../pages/Rocket/Rocket';
import Books from '../components/Books/Books';
import store from '../redux/configureStore';

describe('Test Rocket component ', () => {
  test('Does Books match the snapshot ?', () => {
    const book = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Books />
        </Provider>
      </BrowserRouter>,
    );
    expect(book).toMatchSnapshot();
  });
});
