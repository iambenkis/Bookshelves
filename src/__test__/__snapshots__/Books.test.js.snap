// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rocket component  Does Books match the snapshot ? 1`] = `
<div
  className="book-container"
>
  <div
    className="books"
  />
</div>
`;
